# 1. zadatak

In [ ]:
from sklearn.datasets import fetch_openml
baza = fetch_openml('ilpd')

/usr/local/lib/python3.7/dist-packages/sklearn/datasets/_openml.py:421: UserWarning: Multiple active versions of the dataset matching the name ilpd exist. Versions may be fundamentally different, returning version 1.
  " {version}.".format(name=name, version=res[0]["version"])


In [ ]:
baza.data

V1      V2    V3   V4     V5    V6     V7   V8   V9   V10
0    65.0  Female   0.7  0.1  187.0  16.0   18.0  6.8  3.3  0.90
1    62.0    Male  10.9  5.5  699.0  64.0  100.0  7.5  3.2  0.74
2    62.0    Male   7.3  4.1  490.0  60.0   68.0  7.0  3.3  0.89
3    58.0    Male   1.0  0.4  182.0  14.0   20.0  6.8  3.4  1.00
4    72.0    Male   3.9  2.0  195.0  27.0   59.0  7.3  2.4  0.40
..    ...     ...   ...  ...    ...   ...    ...  ...  ...   ...
578  60.0    Male   0.5  0.1  500.0  20.0   34.0  5.9  1.6  0.37
579  40.0    Male   0.6  0.1   98.0  35.0   31.0  6.0  3.2  1.10
580  52.0    Male   0.8  0.2  245.0  48.0   49.0  6.4  3.2  1.00
581  31.0    Male   1.3  0.5  184.0  29.0   32.0  6.8  3.4  1.00
582  38.0    Male   1.0  0.3  216.0  21.0   24.0  7.3  4.4  1.50

[583 rows x 10 columns]

In [ ]:
podaci = baza.data

In [ ]:
import numpy as np
podaci = np.array(baza.data)
X = podaci[:480, 6:10]

In [ ]:
y = baza.target[:480]
y = y.astype('float32')

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.70)

In [ ]:
from sklearn import svm
model = svm.SVC()
model.fit(X_train, y_train)

SVC()

In [ ]:
y_pred = model.predict(X_test)

In [ ]:
from sklearn.metrics import precision_score, recall_score, f1_score  
print("Precision score: {}".format(precision_score(y_test, y_pred, average=None)))
print("Recall score: {}".format(recall_score(y_test, y_pred, average=None)))
print("F1 score: {}".format(f1_score(y_test, y_pred, average=None)))

Precision score: [0.72916667 0.        ]
Recall score: [1. 0.]
F1 score: [0.84337349 0.        ]


/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [ ]:
X1 = podaci[:, 2:10]

In [ ]:
y1 = baza.target
y1 = y1.astype('float32')

In [ ]:
X1_train, X1_test, y1_train, y1_test = train_test_split(X1, y1, train_size=0.70)

In [ ]:
model1 = svm.SVC()
model1.fit(X1_train, y1_train)

SVC()

In [ ]:
y1_pred = model1.predict(X1_test)

In [ ]:
print("Precision score: {}".format(precision_score(y1_test, y1_pred, average=None)))
print("Recall score: {}".format(recall_score(y1_test, y1_pred, average=None)))
print("F1 score: {}".format(f1_score(y1_test, y1_pred, average=None)))

Precision score: [0.74285714 0.        ]
Recall score: [1. 0.]
F1 score: [0.85245902 0.        ]


/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Drugi model postigao je bolje rezultate.
Veća je preciznost (0.74 naspram 0.73), odaziv je isti.
F1 score naravno također bolji (0.85 naspram 0.84).